### PR TITLE
fix: use token from first room info to build conversationId for sogs

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/session/apis/open_group_api/opengroupV2/JoinOpenGroupV2.ts [hunk 1]
```diff
@@ -154,25 +159,25 @@ export async function joinOpenGroupV2WithUIEvents(
 
     uiCallback?.({ loadingState: 'started', conversationKey: conversationID });
 
-    await joinOpenGroupV2(parsedRoom, fromConfigMessage);
+    const convoCreated = await joinOpenGroupV2(parsedRoom, fromConfigMessage);
 
-    const isConvoCreated = getConversationController().get(conversationID);
+    const isConvoCreated = Boolean(convoCreated);
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/opengroupV2/JoinOpenGroupV2.ts:164]: Do we need the variable assignment here? Can we not just put it in the if statement?

File: ts/session/apis/open_group_api/opengroupV2/OpenGroupManagerV2.ts [hunk 2]
```diff
@@ -163,25 +163,32 @@ export class OpenGroupManagerV2 {
...
         serverPubkey: serverPublicKey,
         serverUrl,
         roomId,
       });
-      if (!roomInfos) {
+
+      if (!roomInfos || !roomInfos.id) {
         throw new Error('Invalid open group roomInfo result');
       }
+      await OpenGroupData.removeV2OpenGroupRoom(conversationId);
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/opengroupV2/OpenGroupManagerV2.ts:186]: So we make one attempt using the conversationId and if that doesn't work we fallback to the roomId?

File: ts/session/apis/open_group_api/opengroupV2/JoinOpenGroupV2.ts (hunk 1 above)
Comment (Bilb) [reply ts/session/apis/open_group_api/opengroupV2/JoinOpenGroupV2.ts:164]: fixed

Comment (yougotwill) [reply ts/session/apis/open_group_api/opengroupV2/JoinOpenGroupV2.ts:164]: 👍

File: ts/session/apis/open_group_api/opengroupV2/OpenGroupManagerV2.ts (hunk 2 above)
Comment (yougotwill) [reply ts/session/apis/open_group_api/opengroupV2/OpenGroupManagerV2.ts:186]: Resolved 👍

**Review by yougotwill — APPROVED** [review]
LGTM 🎸

**Review by yougotwill — APPROVED** [review]

**Pull request merged**